sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionControllerImpl.java

```diff
@@ -109,56 +109,60 @@ public PartitionControllerImpl(
     }
 
     private Mono<Void> removeLease(Lease lease) {
-            if (this.currentlyOwnedPartitions.get(lease.getLeaseToken()) != null) {
-                WorkerTask workerTask = this.currentlyOwnedPartitions.remove(lease.getLeaseToken());
-
+        return Mono.justOrEmpty(this.currentlyOwnedPartitions.remove(lease.getLeaseToken()))
+            .flatMap(workerTask -> {
```

Comment:
change the WorkerTask and add the following method:
`    public void cancelJob() {
        shutdownCts.cancel();
        this.interrupt();
    }
`

Also please change the "done" boolean to an AtomicBoolean or volatile:
`    private volatile boolean done = false;
`

Reply:
originally was doing override for interrupt method, changed to the suggested way: cancelJob. it is clearer.